modules/codemod/lib/v8/deprecateDrawer.ts

```diff
@@ -0,0 +1,105 @@
+import {API, FileInfo, Identifier, ImportDeclaration, Options} from 'jscodeshift';
+
+const mainPackage = '@workday/canvas-kit-labs-react';
+const drawerPackage = '@workday/canvas-kit-labs-react/drawer';
+const drawerImportNames = ['Drawer', 'DrawerProps', 'DrawerHeaderProps', 'DrawerHeader'];
+
+export default function transformer(file: FileInfo, api: API, options: Options) {
+  const j = api.jscodeshift;
+
+  const root = j(file.source);
+  let hasDrawerImports = false;
+
+  // This toggles the failsafe that prevents us from accidentally transforming something unintentionally.
+  root.find(j.ImportDeclaration, (nodePath: ImportDeclaration) => {
+    const value = nodePath.source.value;
+    // If there's an import from the drawer package, set the import boolean check to true
+    if (value === drawerPackage) {
+      hasDrawerImports = true;
+      return false;
+    }
+    // If there's an import from the main package, check to see if Drawer or DrawerProps are among the named imports
+    // e.g. import {Drawer} from '@workday/canvas-kit-react';
+    if (value === mainPackage) {
+      (nodePath.specifiers || []).forEach(specifier => {
+        if (specifier.type === 'ImportSpecifier') {
+          const specifierName = specifier.imported.name;
+          hasDrawerImports = drawerImportNames.includes(specifierName);
+        }
+      });
+    }
+    return false;
+  });
+
+  // Failsafe to skip transforms unless a Drawer import is detected
+  if (!hasDrawerImports) {
+    return root.toSource();
+  }
+
+  root
+    .find(j.ImportDeclaration, {
+      source: {value: (value: string) => value.includes(mainPackage)},
+    })
+    .forEach(nodePath => {
+      nodePath.value.specifiers?.forEach(specifier => {
+        if (specifier.type === 'ImportSpecifier') {
+          //  `import {Drawer}` becomes `import {DeprecatedDrawer}`
+          //  `import {DrawerProps}` becomes `import {DeprecatedDrawerProps}`
+          //  `import {DrawerHeader}` becomes `import {DeprecatedDrawerHeader}`
+          //  `import {DrawerHeaderProps}` becomes `import {DeprecatedDrawerHeaderProps}`
```

Comment:
Does this also account for `DeprecatedDrawerDirection` it's an exported enum

Reply:
Missed that thing! thanks for catching